pkg/controller/controller_instance.go

```diff
@@ -1685,7 +1809,7 @@ func (c *controller) processProvisionSuccess(instance *v1beta1.ServiceInstance,
 	if _, err := c.updateServiceInstanceStatus(instance); err != nil {
 		return err
 	}
-
+	c.removeInstanceFromRetryMap(instance)
```

Comment:
I think there should also be a case when we hit a "terminal" error for a particular generation (e.g. `400 Bad Request`)  and won't retry until the spec is updated. We should invoke `c.removeInstanceFromRetryMap(instance)` in that case as well. Otherwise, when the spec will be updated, we will still retry with accumulated delay...

Reply:
I'm talking about `processTerminalProvisionFailure` (https://github.com/kubernetes-incubator/service-catalog/blob/491f227a4fc7453917e347ccc957da9a312a4774/pkg/controller/controller_instance.go#L1695:22) and `processTerminalUpdateServiceInstanceFailure` (https://github.com/kubernetes-incubator/service-catalog/blob/491f227a4fc7453917e347ccc957da9a312a4774/pkg/controller/controller_instance.go#L1801)

Reply:
Agreed, good catch.